Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransform.java

```diff
@@ -0,0 +1,579 @@
+
+        //*********************************Construct Active Participant************************
+        //Active Participant for human requester only required for requesting gateway
+        if (isRequesting) {
+            AuditMessageType.ActiveParticipant participantHumanFactor = getActiveParticipant(assertion.getUserInfo());
+            auditMsg.getActiveParticipant().add(participantHumanFactor);
+        }
+        AuditMessageType.ActiveParticipant participantSource = getActiveParticipantSource(isRequesting, webContextProperties, serviceName);
+        AuditMessageType.ActiveParticipant participantDestination = getActiveParticipantDestination(target, isRequesting, webContextProperties, serviceName);
+        auditMsg.getActiveParticipant().add(participantSource);
+        auditMsg.getActiveParticipant().add(participantDestination);
+
+        //******************************Constuct Participation Object Identification*************
+        // Assign ParticipationObjectIdentification
+        auditMsg = getParticipantObjectIdentification(request, response, assertion, auditMsg);
+
+        //Create the AuditSourceIdentifierType object
+        String communityId = getMessageCommunityId(assertion, target, isRequesting);
+        AuditSourceIdentificationType auditSource = getAuditSourceIdentificationType();
+
+        //******************************Constuct Audit Source Identification**********************
+        auditMsg.getAuditSourceIdentification().add(auditSource);
+
+        //Set the all the required data
+        result.setAuditMessage(auditMsg);
+        result.setDirection(direction);
+        result.setInterface(_interface);
+        //set the target community identifier
+        result.setCommunityId(communityId);
+
+        LOG.trace("End AuditDataTransform transformMsgToAuditMsg() ----");
+        return result;
+    }
+
+    /**
+     *
+     * @param request
+     * @param response
+     * @param assertion
+     * @param auditMsg
+     * @return
+     */
+    protected abstract AuditMessageType getParticipantObjectIdentification(T request, K response, AssertionType assertion, AuditMessageType auditMsg);
+
+    /**
+     *
+     * @return
+     */
+    protected AuditSourceIdentificationType getAuditSourceIdentificationType() {
+        String hcid = HomeCommunityMap.getLocalHomeCommunityId();
+        String homeCommunityName = HomeCommunityMap.getHomeCommunityName(hcid);
+        return createAuditSourceIdentification(hcid, homeCommunityName);
+    }
+
+    /**
+     *
+     * @param oUserInfo
+     * @return
+     */
+    protected AuditMessageType.ActiveParticipant getActiveParticipant(UserType oUserInfo) {
+        // Create Active Participant Section
+        // create a method to call the AuditDataTransformHelper - one expectation
+        AuditMessageType.ActiveParticipant participant = createActiveParticipantFromUser(
+            oUserInfo, Boolean.TRUE);
```

Comment:
Don't think we need the boolean as a method argument if it's always "TRUE".  Can just set it in the private method.